packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -972,27 +972,43 @@ type RenderCondition = {
   clinicalStatus: string;
 };
 
-function createConditionSection(conditions: Condition[]) {
+function createConditionSection(conditions: Condition[], encounter: Encounter[]) {
   if (!conditions) {
     return "";
   }
 
+  const conditionDateDict = getConditionDatesFromEncounters(encounter);
+
   const removeDuplicate = uniqWith(conditions, (a, b) => {
     const aDate = dayjs(a.onsetDateTime).format(ISO_DATE);
     const bDate = dayjs(b.onsetDateTime).format(ISO_DATE);
-    return aDate === bDate && a.code?.text === b.code?.text;
+    const aText = a.code?.text;
+    const bText = b.code?.text;
+
+    if (aText === undefined || bText === undefined) {
+      return false;
+    }
+    return aDate === bDate && aText === bText;
   })
     .reduce((acc, condition) => {
       const codeName = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE, SNOMED_CODE]);
       const idc10Code = condition.code?.coding?.find(code =>
         code.system?.toLowerCase().includes(ICD_10_CODE)
       );
 
-      const name = idc10Code?.display ?? condition.code?.coding?.[0]?.display ?? "";
+      const name =
+        idc10Code?.display ?? condition.code?.coding?.[0]?.display ?? condition.code?.text ?? "";
       const onsetDateTime = condition.onsetDateTime ?? "";
       const clinicalStatus = condition.clinicalStatus?.coding?.[0]?.code ?? "";
-      const onsetStartTime = condition.onsetPeriod?.start;
-      const onsetEndTime = condition.onsetPeriod?.end;
+      let onsetStartTime = condition.onsetPeriod?.start ?? "";
+      let onsetEndTime = condition.onsetPeriod?.end ?? "";
+
+      if (!onsetStartTime && condition.id) {
+        onsetStartTime = conditionDateDict[condition.id]?.start ?? "";
+      }
+      if (!onsetEndTime && condition.id) {
+        onsetEndTime = condition.onsetPeriod?.end ?? conditionDateDict[condition.id]?.end ?? "";
```

Comment:
NABD, but in theory we also don't need `condition.onsetPeriod?.end` for `onsetEndTime` like we don't do for `onsetStartTime`, right?